test/e2e/gardener/shoot/create_rotate_delete.go

```diff
@@ -22,242 +24,301 @@ import (
-		"app":              "machine",
-		"machine-provider": "local",
-	})).To(Succeed())
-
-	afterStartMachinePodNames := sets.New[string]()
-	for _, item := range afterStartMachinePodList.Items {
-		afterStartMachinePodNames.Insert(item.Name)
-	}
+	ItShouldAnnotateShoot(s, map[string]string{
+		v1beta1constants.GardenerOperation: v1beta1constants.OperationRotateCredentialsStartWithoutWorkersRollout,
+	})
 
-	ExpectWithOffset(1, beforeStartMachinePodNames.Equal(afterStartMachinePodNames)).To(BeTrue())
+	itShouldEventuallyNotHaveOperationAnnotation(s)
 
-	By("Ensure all worker pools are marked as 'pending for roll out'")
-	for _, worker := range f.Shoot.Spec.Provider.Workers {
-		ExpectWithOffset(1, slices.ContainsFunc(f.Shoot.Status.Credentials.Rotation.CertificateAuthorities.PendingWorkersRollouts, func(rollout gardencorev1beta1.PendingWorkersRollout) bool {
-			return rollout.Name == worker.Name
-		})).To(BeTrue(), "worker pool "+worker.Name+" should be pending for roll out in CA rotation status")
+	It("Rotation in preparing without workers rollout status", func(ctx SpecContext) {
+		Eventually(ctx, func(g Gomega) {
+			shootVerifiers.ExpectPreparingWithoutWorkersRolloutStatus(g)
+			utilsverifiers.ExpectPreparingWithoutWorkersRolloutStatus(g)
+		}).Should(Succeed())
+	}, SpecTimeout(time.Minute))
 
-		ExpectWithOffset(1, slices.ContainsFunc(f.Shoot.Status.Credentials.Rotation.ServiceAccountKey.PendingWorkersRollouts, func(rollout gardencorev1beta1.PendingWorkersRollout) bool {
-			return rollout.Name == worker.Name
-		})).To(BeTrue(), "worker pool "+worker.Name+" should be pending for roll out in service account key rotation status")
-	}
+	ItShouldWaitForShootToBeReconciledAndHealthy(s)
 
-	By("Remove last worker pool from spec and check that it is no longer pending for roll out")
-	patch = client.MergeFrom(f.Shoot.DeepCopy())
-	lastWorkerPoolName := f.Shoot.Spec.Provider.Workers[len(f.Shoot.Spec.Provider.Workers)-1].Name
-	f.Shoot.Spec.Provider.Workers = slices.DeleteFunc(f.Shoot.Spec.Provider.Workers, func(worker gardencorev1beta1.Worker) bool {
-		return worker.Name == lastWorkerPoolName
+	It("Ensure workers were not rolled out", func(ctx SpecContext) {
+		Eventually(ctx, func(g Gomega) {
+			shootVerifiers.ExpectWaitingForWorkersRolloutStatus(g)
+			utilsverifiers.ExpectWaitingForWorkersRolloutStatus(g)
+		}).Should(Succeed())
+	}, SpecTimeout(time.Minute))
+
+	It("Compare machine pod names", func(ctx SpecContext) {
+		afterStartMachinePodList := &corev1.PodList{}
+		Eventually(ctx, s.SeedKomega.List(afterStartMachinePodList, client.InNamespace(s.Shoot.Status.TechnicalID), client.MatchingLabels{
+			"app":              "machine",
+			"machine-provider": "local",
+		})).Should(Succeed())
+
+		afterStartMachinePodNames := sets.New[string]()
+		for _, item := range afterStartMachinePodList.Items {
+			afterStartMachinePodNames.Insert(item.Name)
+		}
+
+		Expect(beforeStartMachinePodNames.Equal(afterStartMachinePodNames)).To(BeTrue())
```

Comment:
If this assertion fails, the error will not tell us the difference. Can you rework this assertion?
```suggestion
		Expect(beforeStartMachinePodNames.UnsortedList()).To(ConsistOf(afterStartMachinePodNames.UnsortedList()))
```